File: sample-chat-java/app/src/main/java/com/quickblox/sample/chat/java/ui/activity/AttachmentVideoActivity.java
package com.quickblox.sample.chat.java.ui.activity;

import android.app.DownloadManager;
import android.content.Context;
import android.content.Intent;
import android.media.MediaPlayer;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.os.Environment;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.MediaController;
import android.widget.ProgressBar;
import android.widget.RelativeLayout;
import android.widget.VideoView;

import com.quickblox.sample.chat.java.R;

import java.io.File;

import androidx.annotation.Nullable;

public class AttachmentVideoActivity extends BaseActivity {
    private static final String EXTRA_FILE_NAME = "video_file_name";
    private static final String EXTRA_FILE_URL = "video_file_URL";

    private RelativeLayout rootLayout;
    private VideoView videoView;
    private ProgressBar progressBar;
    private MediaController mediaController;
    private File file = null;

    public static void start(Context context, String attachmentName, String url) {
        Intent intent = new Intent(context, AttachmentVideoActivity.class);
        intent.putExtra(EXTRA_FILE_URL, url);
        intent.putExtra(EXTRA_FILE_NAME, attachmentName);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_show_video);
        initUI();
        loadVideo();
    }

    private void initUI() {
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                getSupportActionBar().setBackgroundDrawable(getDrawable(R.drawable.toolbar_video_player_background));
            }
            getSupportActionBar().setTitle(getIntent().getStringExtra(EXTRA_FILE_NAME));
            getSupportActionBar().setElevation(0);
        }
        rootLayout = findViewById(R.id.layout_root);
        videoView = findViewById(R.id.vv_full_view);
        progressBar = findViewById(R.id.progress_show_video);

        rootLayout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mediaController.show(2000);
            }
        });
    }

    private void loadVideo() {
        progressBar.setVisibility(View.VISIBLE);
        String filename = getIntent().getStringExtra(EXTRA_FILE_NAME);
        File file = new File(getApplication().getFilesDir(), filename);

        if (file != null) {
            mediaController = new MediaController(this);
            mediaController.setAnchorView(videoView);
            videoView.setMediaController(mediaController);
            videoView.setVideoPath(file.getPath());
            videoView.start();
        }

        videoView.setOnPreparedListener(new MediaPlayer.OnPreparedListener() {
            @Override
            public void onPrepared(MediaPlayer mp) {
                progressBar.setVisibility(View.GONE);
                mediaController.show(2000);
            }
        });

        videoView.setOnErrorListener(new MediaPlayer.OnErrorListener() {
            @Override
            public boolean onError(MediaPlayer mp, int what, int extra) {
                progressBar.setVisibility(View.GONE);
                mediaController.hide();
                showErrorSnackbar(R.string.error_load_video, null, new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        loadVideo();
                    }
                });
                return true;
            }
        });
    }

    @Override
    public void onBackPressed() {
        finish();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_activity_video_player, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.menu_player_save:
                saveFileToGallery();
                return true;
            case android.R.id.home:
                onBackPressed();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    private void saveFileToGallery() {
        if (file != null) {
            try {
                String url = getIntent().getStringExtra(EXTRA_FILE_URL);
                DownloadManager.Request request = new DownloadManager.Request(Uri.parse(url));
                request.setDestinationInExternalPublicDir(Environment.DIRECTORY_DOWNLOADS, file.getName());
                request.setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED);
                request.allowScanningByMediaScanner();
                DownloadManager manager = (DownloadManager) getSystemService(Context.DOWNLOAD_SERVICE);
                if (manager != null) {
                    manager.enqueue(request);
                }
            } catch (SecurityException e) {
                if (e.getMessage() != null) {
                    Log.d("Security Exception", e.getMessage());
                }
            }
        }
    }
}

File: sample-chat-kotlin/app/src/main/java/com/quickblox/sample/chat/kotlin/ui/activity/AttachmentVideoActivity.kt
package com.quickblox.sample.chat.kotlin.ui.activity

import android.app.DownloadManager
import android.content.Context
import android.content.Intent
import android.media.MediaPlayer
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.os.Environment
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.MediaController
import android.widget.ProgressBar
import android.widget.RelativeLayout
import android.widget.VideoView
import com.quickblox.sample.chat.kotlin.R
import java.io.File


private const val EXTRA_FILE_NAME = "video_file_name"
private const val EXTRA_FILE_URL = "video_file_URL"

class AttachmentVideoActivity : BaseActivity() {

    private lateinit var rootLayout: RelativeLayout
    private lateinit var videoView: VideoView
    private lateinit var progressBar: ProgressBar
    private lateinit var mediaController: MediaController
    private var file: File? = null

    companion object {
        fun start(context: Context, attachmentName: String?, url: String?) {
            val intent = Intent(context, AttachmentVideoActivity::class.java)
            intent.putExtra(EXTRA_FILE_URL, url)
            intent.putExtra(EXTRA_FILE_NAME, attachmentName)
            context.startActivity(intent)
        }
    }

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_show_video)
        initUI()
        loadVideo()
    }

    private fun initUI() {
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            supportActionBar?.setBackgroundDrawable(getDrawable(R.drawable.toolbar_video_player_background))
        }
        supportActionBar?.title = intent.getStringExtra(EXTRA_FILE_NAME)
        supportActionBar?.elevation = 0f
        rootLayout = findViewById(R.id.layout_root)
        videoView = findViewById(R.id.vv_full_view)
        progressBar = findViewById(R.id.progress_show_video)

        rootLayout.setOnClickListener {
            mediaController.show(2000)
        }
    }

    private fun loadVideo() {
        progressBar.visibility = View.VISIBLE
        val fileName = intent.getStringExtra(EXTRA_FILE_NAME)
        file = File(application.filesDir, fileName)

        file?.let {
            mediaController = MediaController(this)
            mediaController.setAnchorView(videoView)
            videoView.setMediaController(mediaController)
            videoView.setVideoPath(it.path)
            videoView.start()
        }
        videoView.setOnPreparedListener {
            progressBar.visibility = View.GONE
            mediaController.show(2000)
        }

        videoView.setOnErrorListener(object : MediaPlayer.OnErrorListener {
            override fun onError(mp: MediaPlayer?, what: Int, extra: Int): Boolean {
                progressBar.visibility = View.GONE
                mediaController.hide()
                showErrorSnackbar(R.string.error_load_video, null, object : View.OnClickListener {
                    override fun onClick(v: View?) {
                        loadVideo()
                    }
                })
                return true
            }
        })
    }

    override fun onBackPressed() {
        finish()
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_activity_video_player, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.menu_player_save -> {
                saveFileToGallery()
                return true
            }
            android.R.id.home -> {
                onBackPressed()
                return true
            }
            else -> return super.onOptionsItemSelected(item)
        }
    }

    private fun saveFileToGallery() {
        if (file != null) {
            try {
                val url = intent.getStringExtra(EXTRA_FILE_URL)
                val request = DownloadManager.Request(Uri.parse(url))
                request.setDestinationInExternalPublicDir(Environment.DIRECTORY_DOWNLOADS, file!!.name)
                request.setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED)
                request.allowScanningByMediaScanner()
                val manager = getSystemService(Context.DOWNLOAD_SERVICE) as DownloadManager
                manager.enqueue(request)
            } catch (e : SecurityException) {
                Log.d("Security Exception", e.message)
            }
        }
    }
}